### PR TITLE
Migrate files sync related jobs to WorkManager API

### LINKED ISSUE #5881: Content observer work is not re-scheduled after triggered by media update
### Steps to reproduce
1. Configure instant upload of camera images
2. take a photo few times

### Expected behaviour
- Instant upload should be triggered shortly after taking a photo

### Actual behaviour
- Nothing happens - media does not trigger content observer task

### PR DESCRIPTION
**Pull request by ezaquarii**
* FilesSyncJob.java converted to FilesSyncWork.kt
* FilesSyncJob removed from code
* Extended ETM screen to allow scheduling immediate
  TestJob (for development experimentation)
* OfflineSyncJob.java conveted to OfflineSyncWork.kt
* OfflineSyncJob removed from code
* Fixed re-scheduling of content observer work
* MediaFoldersDetectionJob.java converted to
  MediaFoldersDetectionWork.kt
* MediaFolderDetectionJob removed from code

Fixes #5881

Signed-off-by: Chris Narkiewicz <hello@ezaquarii.com>

### CODE REVIEW

File: src/main/java/com/nextcloud/client/jobs/FilesSyncWork.kt [hunk 1]
```diff
@@ -0,0 +1,234 @@
...
+import com.owncloud.android.utils.MimeType
+import com.owncloud.android.utils.MimeTypeUtil
+import java.io.File
+import java.text.ParsePosition
+import java.text.SimpleDateFormat
+import java.util.Locale
+import java.util.TimeZone
+
+@Suppress("LongParameterList") // legacy code
+class FilesSyncWork(
```
Comment (ezaquarii) [inline src/main/java/com/nextcloud/client/jobs/FilesSyncWork.kt:63]: This is `FilesSyncJob.java` auto-converted to Kotlin + few minor stylisting tweaks to satisfy `Detekt` and `Ktlint`. There is no functional change relative to old job logic.

**Comment by ezaquarii** [comment]
1. Fixed failing test.
2. Rebased on top of recent `FilesSyncJob` changes.

**Comment by ezaquarii** [comment]
@tobiasKaminsky I ported the fix from #5885 - it uses job manager instead of helper but should otherwise be semantically indentical.

**Comment by ezaquarii** [comment]
rebased to resolve findbugs txt conflict

**Comment by tobiasKaminsky** [comment]
Nice @ezaquarii as always :tada: 
Let us merge this, so we have maximum time until next 3.13 for testing :+1:

**Pull request merged**